tpke/src/lib.rs

```diff
@@ -318,4 +439,69 @@ mod tests {
         let aad = "bad aad".as_bytes();
         assert!(!check_ciphertext_validity(&ciphertext, aad));
     }
+
+    #[test]
+    fn simple_threshold_decryption() {
+        let mut rng = &mut test_rng();
+        let threshold = 16 * 2 / 3;
+        let shares_num = 16;
+        let msg: &[u8] = "abc".as_bytes();
+        let aad: &[u8] = "my-aad".as_bytes();
+
+        // To be updated
+        let (pubkey, _privkey, private_decryption_contexts) =
+            setup_simple::<E>(threshold, shares_num, &mut rng);
+
+        // Stays the same
+        // Ciphertext.commitment is already computed to match U
+        let mut ciphertext = encrypt::<_, E>(msg, aad, &pubkey, rng);
+
+        // Creating decryption shares
+        let decryption_shares = private_decryption_contexts
+            .iter()
+            .map(|context| {
+                let u = ciphertext.commitment;
+                let z_i = context.private_key_share.clone();
+                // Simplifying to just one key share per node
+                assert_eq!(z_i.private_key_shares.len(), 1);
+                let z_i = z_i.private_key_shares[0];
+                // Really want to call E::pairing here to avoid heavy computations on client side
+                // C_i = e(U, Z_i)
+                // TODO: Check whether blinded key share fits here
+                E::pairing(u, z_i)
```

Comment:
This assumes the blinded key share is already "unblinded"

Reply:
Yes. I think I need to revisit PVSS implementation in `ferveo` and make sure that blinding makes sense on the `tpke` level.